src/sentry/tasks/base.py

```diff
@@ -23,8 +23,14 @@ def _wrapped(*args, **kwargs):
                 instance = '{}.{}'.format(name, stat_suffix(*args, **kwargs))
             else:
                 instance = name
-            with metrics.timer(key, instance=instance):
-                result = func(*args, **kwargs)
+            with metrics.timer(key, instance=instance) as tags:
+                try:
+                    result = func(*args, **kwargs)
+                except Exception:
+                    tags['result'] = 'failure'
+                    raise
```

Comment:
I don't think this does anything for python 2: https://bitbucket.org/gutworth/six/src/8a545f4e906f6f479a6eb8837f31d03731597687/six.py?at=default&fileviewer=file-view-default#six.py-701:703

So this is prob fine.